src/main/java/bio/terra/common/iam/AuthenticatedUserRequest.java

```diff
@@ -40,6 +58,95 @@ public String getToken() {
     return token;
   }
 
+  public FlightMap putIn(FlightMap flightMap, String key) {
+    return putRecord(
+        flightMap,
+        key,
+        AuthenticatedUserRequestModel.getClassSchema(),
+        toModel(),
+        AuthenticatedUserRequestModel.class);
+  }
+
+  public static AuthenticatedUserRequest getFrom(FlightMap flightMap, String key) {
+    AuthenticatedUserRequestModel model =
+        getRecord(
+            flightMap,
+            key,
+            AuthenticatedUserRequestModel.getClassSchema(),
+            AuthenticatedUserRequestModel.class);
+    return fromModel(model);
+  }
+
+  /**
+   * This could be put on FlightMap as a part of Stairway. Broken out to a separate function here to
+   * illustrate.
+   */
+  @VisibleForTesting
+  static <T> FlightMap putRecord(
+      FlightMap flightMap, String key, Schema schema, T record, Class<T> type) {
+    try {
+
+      // Encode the object with the "writer" schema into the byte stream.
+      OutputStream stream = new ByteArrayOutputStream();
+      Encoder jsonEncoder = EncoderFactory.get().jsonEncoder(schema, stream);
```

Comment:
any particular reason to use the json encoder instead of the binary encoder?

Reply:
As one would expect, binary encoded data looks like a lot like protobuf:
```
 test@example.com\u000ESubject 0123.456-789AbCd\u0000\u0006foo
```

The big downside here is that "writer schema" must be stored with the data, and is JSON only; Avro is designed to be used either with large data sets where the schema is stored in a header with many records, or using a [schema registry](https://www.confluent.io/product/confluent-platform/data-compatibility) where a fingerprint is used as a key.  We could do our own schema management, but this seems like a lot of added complexity, and kind of moves backwards on the tight-coupling we're trying to get away from (assuming the schemas get stored in the database).

On the positives, I do like the no-code-generation option (see other PR); if we were to go with Avro with code generation I suspect there's a bit of work we'd want to do to tweak the [Velocity templates](https://velocity.apache.org/) that Avro uses to generate the classes

Reply:
Btw, there may also be no-code-generation options for protobuf: https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/DynamicMessage

Reply:
Ha, I actually really dislike the no-code-generation options. Give me generated classes and type safety where I can't fat finger the field name.

Reply:
In general I agree, but a bad object mode for the generated code can be a downside, which I think is reflected in the semantics of   

Protobuf seems to have been pretty thoughtfully designed their Java object model.  Avro can achieve something similar, but would require more customization to get there.  Out of the box the FlightMap-facing interface is awkward.